### PR TITLE
Implemented lateness functionality + some more refactoring

### PR DESCRIPTION
**Pull request by mvoutov**
Note to anyone reviewing this: this pull request has implemented the new package into our app _and_ the lateness modal with proper formatting; we can upload and download YAML policy files and grades and update the lateness policies and categories

### CODE REVIEW

**Review by calv2n — APPROVED** [review]
Great! Only concern is maybe www/ directory should live outside of the R/ directory.

File: R/server.R [hunk 1]
```diff
@@ -20,45 +21,109 @@ shinyServer(function(input, output, session) {
...
         tryCatch({
-            uploaded_data <- gradebook::read_gs(input$upload_gs$datapath)
+            uploaded_data <- read_csv(input$upload_gs$datapath)
             data(uploaded_data)
         }, error = function(e) {
             showNotification('Please upload a file with the Gradescope format','',type = "error")
             
         })
     })
+
```
Comment (andrewpbray) [inline R/server.R:31]: Is there a reason that this isn't just deleted? Keep in mind with git, it's very easy to access this code from previous commits if we want to re-incorporate it.

File: R/ui.R [hunk 2]
```diff
@@ -35,9 +35,9 @@ shinyUI(
                 ),
                 fileInput("upload_policy", label = NULL, accept = c(".yml")),
                 hr(),
-              h5("Explore With Generic Data", style = 'margin-left: 20px;'),
-              div(actionButton('demogs', "Use Demo Data and Policy")),
-              hr(),
+              # h5("Explore With Generic Data", style = 'margin-left: 20px;'),
```
Comment (andrewpbray) [inline R/ui.R:38]: Ditto: should this just be deleted?

**Review by andrewpbray — COMMENTED** [review]
Wow, there was a ton a work that went into this. Really wonderful work you two!

**Review by andrewpbray — APPROVED** [review]

**Comment by mvoutov** [comment]
> Great! Only concern is maybe www/ directory should live outside of the R/ directory.

@calv2n thanks for pointing this out; I believe this is the correct place for it. 

image

**Comment by ZBallDiesel** [comment]
@andrewpbray Sure! I reviewed the changes and the app is looking very nice!

**Review by ZBallDiesel — COMMENTED** [review]
Very good work everyone!

**Pull request merged**